lms/models/canvas_file.py

```diff
@@ -0,0 +1,44 @@
+import sqlalchemy as sa
+from sqlalchemy.dialects.postgresql import ARRAY
+from sqlalchemy.ext.mutable import MutableList
+
+from lms.db import BASE
+
+
+class CanvasFile(BASE):
+    """
+    A file that we've seen in a response from the Canvas Files API.
+
+    https://canvas.instructure.com/doc/api/files.html
+    """
+
+    __tablename__ = "canvas_file"
+    __table_args__ = (
+        sa.UniqueConstraint("consumer_key", "tool_consumer_instance_guid", "file_id"),
+    )
+
+    id = sa.Column(sa.Integer, primary_key=True)
```

Comment:
Following our usual practice I added a "surrogate key" (the `id` column) in addition to the "natural key" provided by the unique constraint on `consumer_key`, `tool_consumer_instance_guid`, `file_id`.

This means that if we ever have another table that wants to have a foreign key to the `canvas_file` table, the other table will only need a single `canvas_file_id` column rather than `consumer_key`, `tool_consumer_instance_guid`, and `file_id` columns.

The integer primary key also provides a total ordering over the `canvas_file` rows, and in the order they were added too. Can be useful for tasks that want to iterate over the whole table.